cmd/cli/trafficpolicy_check.go

```diff
@@ -0,0 +1,205 @@
+	if permissiveMode, err := cmd.isPermissiveModeEnabled(); err != nil {
+		return errors.Errorf("Error checking if permissive mode is enabled")
+	} else if permissiveMode {
+		fmt.Fprintf(cmd.out, "[+] Permissive mode enabled for mesh operated by osm-controller running in %s namespace\n\n "+
+			"[+] Pod '%s/%s' is allowed to communicate to pod '%s/%s'\n",
+			osmNamespace, srcPod.Namespace, srcPod.Name, dstPod.Namespace, dstPod.Name)
+		return nil
+	}
+
+	// SMI traffic policy mode
+	fmt.Fprintf(cmd.out, "[+] SMI traffic policy mode enabled for mesh operated by osm-controller running in %s namespace\n\n", osmNamespace)
+	trafficTargets, err := cmd.smiAccessClient.AccessV1alpha2().TrafficTargets(dstPod.Namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return errors.Errorf("Error listing SMI TrafficTarget policies: %s", err)
+	}
+
+	var foundTrafficTarget bool
+	for _, trafficTarget := range trafficTargets.Items {
+		spec := trafficTarget.Spec
+		// Map traffic targets to the given pods
+		if spec.Destination.Name == dstPod.Spec.ServiceAccountName && spec.Destination.Namespace == dstPod.Namespace {
+			// The TrafficTarget destination is associated to 'dstPod'
+
+			// Check if 'srcPod` is an allowed source to this destination
+			for _, source := range spec.Sources {
+				if source.Name == srcPod.Spec.ServiceAccountName && source.Namespace == srcPod.Namespace {
+					fmt.Fprintf(cmd.out, "[+] Pod '%s/%s' is allowed to communicate to pod '%s/%s' via the SMI TrafficTarget policy %q:\n",
+						srcPod.Namespace, srcPod.Name, dstPod.Namespace, dstPod.Name, trafficTarget.Name)
+					foundTrafficTarget = true
+
+					target := trafficTarget // avoids gosec G601: Implicit memory aliasing in for loop
+					trafficTargetPolicy, err := yaml.Marshal(&target)
+					if err != nil {
+						return errors.Errorf("Failed to marshal TrafficTarget %s: %s", trafficTarget.Name, err)
+					}
+					fmt.Fprintf(cmd.out, "---\n%s\n---\n", string(trafficTargetPolicy))
+				}
+			}
+		}
+	}
+
+	if !foundTrafficTarget {
+		fmt.Fprintf(cmd.out, "[+] Pod '%s/%s' is not allowed to communicate to pod '%s/%s', missing SMI TrafficTarget policy\n",
+			srcPod.Namespace, srcPod.Name, dstPod.Namespace, dstPod.Name)
+	}
+
+	return nil
+}
+
+func (cmd *trafficPolicyCheckCmd) getMeshedPod(namespace, podName string) (*corev1.Pod, error) {
+	// Validate the pods
+	pod, err := cmd.clientSet.CoreV1().Pods(namespace).Get(context.TODO(), podName, metav1.GetOptions{})
+	if err != nil {
+		return nil, errors.Errorf("Could not find pod %s in namespace %s", podName, namespace)
+	}
+	if !isMeshedPod(*pod) {
+		return nil, errors.Errorf("Pod %s in namespace %s is not a part of a mesh", podName, namespace)
+	}
+	return pod, nil
+}
+
+func (cmd *trafficPolicyCheckCmd) isPermissiveModeEnabled() (bool, error) {
+	osmNamespace := settings.Namespace()
+	configMap, err := cmd.clientSet.CoreV1().ConfigMaps(osmNamespace).Get(context.TODO(), osmConfigMapName, metav1.GetOptions{})
```

Comment:
We have an opportunity to reuse `pkg/configurator`
Do you think you could carve out this: https://github.com/openservicemesh/osm/blob/3af31b59f7dfaacb76e480f35cc9ae6cab2b44a1/pkg/configurator/client.go#L138-L156
into a separate function and use it here?

You already have `configMap` here - could pass it to a new function which contains the code from the link above and then use `.PermissiveTrafficPolicyMode`: https://github.com/openservicemesh/osm/blob/3af31b59f7dfaacb76e480f35cc9ae6cab2b44a1/pkg/configurator/methods.go#L41-L43

Reply:
This needs a bit more refactoring, created an issue to reuse code: https://github.com/openservicemesh/osm/issues/1962